pkg/controller/awsfederatedaccountaccess/awsfederatedaccountaccess_controller.go

```diff
@@ -768,3 +753,37 @@ func hasLabel(awsFederatedAccountAccess *awsv1alpha1.AWSFederatedAccountAccess,
 	}
 	return false
 }
+
+func checkAndDeletePolicy(reqLogger logr.Logger, awsClient awsclient.Client, uidLabel string, crPolicyName string, policyName *string, policyArn *string) error {
+
+	var awsCustomPolicyname string
+	awsCustomPolicyname = checkUidLabel(awsCustomPolicyname, crPolicyName, uidLabel)
```

Comment:
With your current implementation of `checkUidLabel` you'd always be checking `if "" != crPolicyName+"-"+uidLabel`, so this check is irrelevant.  

What we discussed in our sync is checking f the value in `crPolicyName` already contains the `"-"+uidLabel` postfix. Ideally this function should check `crPolicyName` to see if it already contains the uidLabel, if so, return `crPolicyName` as is, if not, return `crPolicyName+"-"+uidLabel`. If any of the above isn't clear we can sync up again to discuss :)